src/main/CommandLine.cpp

```diff
@@ -1360,6 +1360,48 @@ getSettingsUpgradeTransactions(CommandLineArgs const& args)
         });
 }
 
+int
+runCheckQuorumIntersection(CommandLineArgs const& args)
+{
+    CommandLine::ConfigOption configOption;
+    std::string jsonPath;
+    return runWithHelp(
+        args, {configurationParser(configOption), fileNameParser(jsonPath)},
```

Comment:
It seems that a configuration file should not be necessary to check the provided json file? If that's the case then it might be better not to require `--conf`.

Reply:
You're right that the algorithm itself doesn't need the config, but `QuorumIntersectionCheckerImpl` requires one and uses it to generate human readable names (if available). I'll try to make the config optional instead.

Reply:
I just pushed up [a commit](https://github.com/stellar/stellar-core/pull/4097/commits/32a690e220403b367dfca7b62355c908a6ea7a7f) that makes `--conf` optional.

Reply:
Thanks! It looks like it's trying load a default config file though:
```
./src/stellar-core check-quorum-intersection src/testdata/test-network.json 
2023-12-19T15:54:58.703 [default INFO] Config from stellar-core.cfg
2023-12-19T15:54:58.703 [SCP FATAL] check-quorum-intersection encountered an error: No config file stellar-core.cfg found
```

Reply:
Whoops, I forgot to actually mark it optional in this case. *now* it should work with 2408801b6410d31d02f261dd82e65907e8a5555e

Reply:
Alternatively, it seems cleaner to just make the Config argument in QuorumIntersectionChecker optional, then we don't need to construct the default config at all (I think it's also a bit error-prone to pass default config to a module that is expecting the real user config).

Reply:
Good idea. I changed the `Config` in `QuorumIntersectionChecker` to a `std::optional<Config>` changed the front end to pass `std::nullopt` when `--conf` isn't passed (see 31dcd9a31c448fdc749895aacf74b8649077e523 for details)